src/app/CASESessionManager.cpp

```diff
@@ -54,6 +55,14 @@ void CASESessionManager::FindOrEstablishSession(const ScopedNodeId & peerId, Cal
             {
```

Comment:
I think it could be worthwhile to do a conditional here rather than calling both, i.e. if (onExtendedCOnnectionFailure != null) {} else { existing code }

I think also treating this as a oneof, i.e. not letting callers pass both, is worthwhile

Reply:
Acked